### PR TITLE
School Library: Refactor code

### PR DESCRIPTION
**Pull request by Emmy-github-webdev**
# School Library: Refactor code

In this project @Ntare22 and @Emmy-github-webdev 
- Ensure Solid principles are followed
- Ensure Ruby style guide, YAGNI, KISS, and DRY

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -90,7 +90,6 @@ def create_a_student
     student = Student.new(classroom: @classroom, age: age, name: name, parent_permission: has_permission)
     @people.push(student)
     puts 'Student created successfully'
-    puts
     choose_a_number
   end
```
Comment (mikethreels) [inline app.rb]: - [ ] in the `create_a_student` method you are printing to the console getting information from the user and saving the information this is not in line with the single responsibility rule, have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for more information.

File: app.rb [hunk 2]
```diff
@@ -104,7 +103,6 @@ def create_a_teacher
     teacher = Teacher.new(age: age, name: name, specialization: specialization)
     @people.push(teacher)
     puts 'Teacher created successfully'
-    puts
     choose_a_number
   end
```
Comment (mikethreels) [inline app.rb]: - [ ] in the `create_a_teacher` method you are printing to the console getting information from the user and saving the information this is not in line with the single responsibility rule, have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for more information.

File: app.rb [hunk 3]
```diff
@@ -116,7 +114,6 @@ def create_a_book
     book = Book.new(title, author)
     @book.push(book)
     puts 'Book cretaed successfully'
-    puts
     choose_a_number
   end
```
Comment (mikethreels) [inline app.rb]: - [ ] in the `create_a_book` method you are printing to the console getting information from the user and saving the information this is not in line with the single responsibility rule, have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for more information.

File: app.rb [hunk 4]
```diff
@@ -140,7 +137,6 @@ def create_a_rental
 
     puts 'Rental created successfully!'
     choose_a_number
-    puts
   end
 
   def list_all_rentals_for_a_person
```
Comment (mikethreels) [inline app.rb:148]: - [ ] in the `create_a_rental` method you are printing to the console getting information from the user and saving the information this is not in line with the single responsibility rule, have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for more information.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Emmyn5600 — APPROVED** [review]
STATUS: APPROVED :partying_face: 

Wooow👏, Your project is complete!  There is nothing else to say other than... it's time to merge it. 👏

Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**